src/kudu/consensus/raft_consensus.cc

```diff
@@ -3465,7 +3479,8 @@ void RaftConsensus::UpdateFailureDetectorState(boost::optional<MonoDelta> delta)
   DCHECK(lock_.is_locked());
   const auto& uuid = peer_uuid();
   if (uuid != cmeta_->leader_uuid() &&
-      cmeta_->IsVoterInConfig(uuid, ACTIVE_CONFIG)) {
+      cmeta_->IsVoterInConfig(uuid, ACTIVE_CONFIG) &&
+      !persistent_vars_->is_failure_detector_disabled()) {
```

Comment:
I think this line belongs in EnableFailureDetector(), i.e. return no-op if disabled and treat it in the same way as the gflag check